### ISSUE TITLE
Sending E2EE messages during app startup can wedge all E2EE sending.

### ISSUE THREAD

**Issue by ara4n** [vector-im/element-web#16194]
While my app (Nightly) was starting, I tried to queue up some E2EE messages to be sent (during the ~10 minutes it takes the app to launch, as per https://github.com/vector-im/element-web/issues/13266).  In practice, something silently failed while setting up the new megolm session for the message, wedging all subsequent megolm session sharing (and so leaving all messages suck in 'green').

It got as far as:

```
2021-01-18T12:06:24.903Z I Preparing to encrypt events for !sSKGBJDDYUBJnHqVHY:matrix.org
2021-01-18T12:06:24.904Z I downloadKeys: already have all necessary keys
2021-01-18T12:06:24.929Z I Starting new megolm session for room !sSKGBJDDYUBJnHqVHY:matrix.org
```

...and then once the app had synced...

```
2021-01-18T12:12:55.413Z I Storing megolm session iaPkNDav0ZQlg/B5aKjj0LJOOXbhcjlPswIplp9dB0c/JO1zKw3/l4x3k9VQ0g09xoc/EHY2j1584uC3pBQjqOQ with first index 0
2021-01-18T12:12:55.535Z I Started new megolm session JO1zKw3/l4x3k9VQ0g09xoc/EHY2j1584uC3pBQjqOQ for room !sSKGBJDDYUBJnHqVHY:matrix.org
```

but never progressed beyond:
```
2021-01-18T12:12:58.554Z I PUT /sendToDevice/m.room.encrypted/m1610971978554.9 {"@tom:lant.uk":["BRQOYZMEDG","CMMIYEXSUM","NDRJDABOBE","SKHAKWKGQU","TRWUGQBYJH","MNTYOESDAV","FIZTMFNYFB","YZTNZZOWZE"],"@matthew:matrix.org":["KATBTPMIXQ","PGMUYJXWSG","PPMSYYLWBW","ZTLZPGZQXS","XZSFTSZIDS","EDFZYJHPTY"],"@Amandine:matrix.org":["RMDNVPSGIQ","EJTJYWXNSS","QJUIVUEYUV","AYYWBBVECN"]}
2021-01-18T12:12:58.644Z I Completed megolm keyshare for JO1zKw3/l4x3k9VQ0g09xoc/EHY2j1584uC3pBQjqOQ in !sSKGBJDDYUBJnHqVHY:matrix.org (slice 1/1)
```

In other words, it never got to "encrypting message with megolm session...", presumably due to one of the promises in the big `Promise.all()` in _ensureOutboundSession never completing.

I think @lampholder has reported this failure mode in the past, but I can't find the bug anywhere.

**Comment by jryans** [comment]
We can start by adding some more logs in this path, so we have a better idea where it gets stuck next time.

**Comment by jryans** [comment]
Some logs were added in https://github.com/matrix-org/matrix-js-sdk/pull/1580, so hopefully that will help pinpoint the issue.

**Comment by jryans** [comment]
@ara4n retried with the extra logging code, but so far the issue hasn't come back. I'll move this to the backlog until we learn more.

**Comment by jryans** [comment]
@ara4n has confirmed the issue is indeed fixed now. 🎉
